pkg/webhooks/podimagespec/podimagespec.go

```diff
@@ -0,0 +1,281 @@
+package podimagespec
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+
+	"github.com/openshift/managed-cluster-validating-webhooks/pkg/k8sutil"
+	"github.com/openshift/managed-cluster-validating-webhooks/pkg/webhooks/utils"
+
+	imagestreamv1 "github.com/openshift/api/image/v1"
+	configv1 "github.com/openshift/api/imageregistry/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	admissionregv1 "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	admissionctl "sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const (
+	WebhookName string = "podimagespec-mutation"
+	docString   string = `OpenShift debugging tools on Managed OpenShift clusters must be available even if internal image registry is removed.`
+)
+
+var (
+	timeout int32 = 2
+	scope         = admissionregv1.NamespacedScope
+	rules         = []admissionregv1.RuleWithOperations{
+		{
+			Operations: []admissionregv1.OperationType{
+				admissionregv1.Create,
+			},
+			Rule: admissionregv1.Rule{
+				APIGroups:   []string{""},
+				APIVersions: []string{"v1"},
+				Resources:   []string{"pods"},
+				Scope:       &scope,
+			},
+		},
+	}
+	log        = logf.Log.WithName(WebhookName)
+	imageRegex = regexp.MustCompile(`(image-registry\.openshift-image-registry\.svc:5000\/)(?P<namespace>\S*)(/)(?P<image>\S*)(:)(?P<tag>\S*)`)
+)
+
+// PodImageSpecWebhook mutates an image spec in a pod
+type PodImageSpecWebhook struct {
+	s          *runtime.Scheme
+	configV1   configv1.Config
+	imageV1    imagestreamv1.Image
+	kubeClient client.Client
+}
+
+// NewWebhook creates the new webhook
+func NewWebhook() *PodImageSpecWebhook {
+	scheme := runtime.NewScheme()
+	err := configv1.Install(scheme)
+	if err != nil {
+		fmt.Println("Error:", err)
+		return nil
+	}
+	return &PodImageSpecWebhook{
+		s: scheme,
+	}
+}
+
+// CheckImageRegistryStatus checks the status of the image registry service
+func (s *PodImageSpecWebhook) CheckImageRegistryStatus(ctx context.Context) (bool, error) {
+	var err error
+	if s.kubeClient == nil {
+		s.kubeClient, err = k8sutil.KubeClient(s.s)
+		if err != nil {
+			return false, err
+		}
+	}
+
+	err = s.kubeClient.Get(ctx, client.ObjectKey{Name: "cluster"}, &s.configV1)
+	if err != nil {
+		return false, fmt.Errorf("failed to get image registry config: %v", err)
+	}
+
+	if s.configV1.Spec.ManagementState == operatorv1.Managed || s.configV1.Spec.ManagementState == operatorv1.Unmanaged {
+		return true, nil
+	} else if s.configV1.Spec.ManagementState == operatorv1.Removed {
+		return false, nil
```

Comment:
I don't understand why we are returning false if the image-registry has been removed. Shouldn't we be returning true here like we are doing when it's unmanaged?